HARK/core.py

```diff
@@ -611,8 +622,12 @@ def construct(self, *args, force=False):
                     else:
                         raise ValueError("No constructor found for " + key) from None
 
-                # If this constructor is None, do nothing and mark it as completed
+                # If this constructor is None, do nothing and mark it as completed;
+                # this includes restoring the previous value if it exists
                 if constructor is None:
+                    if key in backup.keys():
+                        setattr(self, key, backup[key])
+                        self.parameters[key] = backup[key]
```

Comment:
there are 2 places where the back up is restored

Reply:
Well, it would throw an error unless you passed `force=True`.